rspace-bench/src/test/scala/coop/rchain/rspace/bench/serialization/ModelSerializerBench.scala

```diff
@@ -0,0 +1,175 @@
+package coop.rchain.rspace.bench.serialization
+
+import java.util.concurrent.TimeUnit
+import java.nio.ByteBuffer
+
+import coop.rchain.models._
+import coop.rchain.models.testImplicits._
+import coop.rchain.rspace.internal._
+import org.scalacheck.Gen.Parameters
+import org.scalacheck.rng.Seed
+import coop.rchain.rspace.bench.serialization._
+
+import org.openjdk.jmh.annotations.{State => BenchState, _}
+import org.openjdk.jmh.infra.Blackhole
+
+class ModelSerializerBench {
+
+  @Benchmark
+  @BenchmarkMode(Array(Mode.AverageTime))
+  @OutputTimeUnit(TimeUnit.MILLISECONDS)
+  @Fork(value = 1)
+  @Warmup(iterations = 5)
+  @Measurement(iterations = 10)
+  def protobufRoundTrip(bh: Blackhole, state: ProtobufModelBenchState) = {
+    val gnat = state.gnat
+    val res  = state.roundTrip(gnat)(state.serializer)
+    bh.consume(res)
+  }
+
+  @Benchmark
+  @BenchmarkMode(Array(Mode.AverageTime))
+  @OutputTimeUnit(TimeUnit.MILLISECONDS)
+  @Fork(value = 1)
+  @Warmup(iterations = 5)
+  @Measurement(iterations = 10)
+  def protobufHeavyRoundTrip(bh: Blackhole, state: ProtobufModelBenchState) = {
+    val res = state.roundTripMany(state.heavyGnats)(state.serializer)
+    bh.consume(res)
+  }
+
+  @Benchmark
+  @BenchmarkMode(Array(Mode.AverageTime))
+  @OutputTimeUnit(TimeUnit.MILLISECONDS)
+  @Fork(value = 1)
+  @Warmup(iterations = 5)
+  @Measurement(iterations = 10)
+  def kryoRoundTrip(bh: Blackhole, state: KryoModelBenchState) = {
+    val gnat = state.gnat
+    val res  = state.roundTrip(gnat)(state.serializer)
+    bh.consume(res)
+  }
+
+  @Benchmark
+  @BenchmarkMode(Array(Mode.AverageTime))
+  @OutputTimeUnit(TimeUnit.MILLISECONDS)
+  @Fork(value = 1)
+  @Warmup(iterations = 5)
+  @Measurement(iterations = 10)
+  def kryoHeavyRoundTrip(bh: Blackhole, state: KryoModelBenchState) = {
+    val res = state.roundTripMany(state.heavyGnats)(state.serializer)
+    bh.consume(res)
+  }
+
+}
+
+abstract class ModelSerializerBenchState {
+  type TestGNAT = GNAT[Par, BindPattern, ListParWithRandom, TaggedContinuation]
+
+  implicit def serializer: Serialize2ByteBuffer[TestGNAT]
+
+  def roundTrip[A](in: A)(implicit s: Serialize2ByteBuffer[A]): A = {
+    val meta = s.encode(in)
+    val out  = s.decode(meta)
+    assert(out == in)
+    out
+  }
+
+  def roundTripMany[A](seq: Seq[A])(implicit s: Serialize2ByteBuffer[A]): Seq[A] =
+    seq.map(roundTrip(_))
+
+  import collection.immutable.Seq
+
+  import org.scalacheck._
+  import org.scalacheck.Arbitrary._
+
+  import coop.rchain.rspace.Serialize
+
+  implicit def arbitraryDatum[C, T](chan: C)(
```

Comment:
+1 for doing the arbitraries by hand. If they were derived by shapeless, they'd include nonsense GNATs, like no patterns and 10 channels, or similar. Which might be OK for serialization tests, but could skew benchmark results